### ISSUE TITLE
BigNum casting issue while parsing Hash into json

### ISSUE THREAD

**Issue by anup1710** [jruby/jruby#5690]
### Environment

Provide at least:
* JRuby version - 9.2.7.0
* Operating system and platform: Windows-64bit

Other relevant info you may wish to add:
* Application/framework version: Sinatra

```ruby
require 'json'
JSON({"a": 1, "b": 23886820000238866000})
```

### Expected Behavior

* JRuby hash should be parsed without raising casting exception for long integer.
* Same problem was raised in json repository. But seems like issue is not fully fixed - https://github.com/flori/json/issues/336

## Actual Behavior

Refer my last comment in json repository issue which has stack trace and exception details.

**Comment by headius** [comment]
Oops, I knew we forgot to update something in 9.2.7.

You should be able to install the gem and use it. We'll include the update in 9.2.8.

**Comment by headius** [comment]
My mistake, we DID include json 2.2.0 in JRuby 9.2.7. Hmmm.

**Comment by headius** [comment]
I cannot reproduce; this appears to work fine with 9.2.7.0. Are you sure you're running the right JRuby?

```
[] ~/projects/jruby $ cat blah.rb
require 'json'
JSON({"a": 1, "b": 23886820000238866000})

[] ~/projects/jruby $ rvm jruby-9.2.7.0 do ruby blah.rb

[] ~/projects/jruby $ 
```

**Comment by headius** [comment]
@anup1710 Your screenshot of the error clearly shows you're running 9.2.6.0, so I think perhaps you should double check your environment. This is fixed in 9.2.7.0.

**Comment by anup1710** [comment]
> @anup1710 Your screenshot of the error clearly shows you're running 9.2.6.0, so I think perhaps you should double check your environment. This is fixed in 9.2.7.0.

My bad... It was typo in jruby version instead of 9.2.6.0 I have written 9.2.7.0.
Thanks for your time and investigation. I will check with latest jruby version.
